Pointers/UniquePtr.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <iostream>
+
+using std::cout;
+using std::endl;
+
+template<typename T>
+class UniquePtr {
+public:
+	UniquePtr(T* ptr);
+	~UniquePtr();
+	// Don't allow copy
+	UniquePtr(const UniquePtr<T>& other) = delete;
+	UniquePtr<T>& operator=(const UniquePtr<T>& other) = delete;
+	// Allow move
+	UniquePtr(UniquePtr<T>&& other) noexcept;
+	UniquePtr<T>& operator=(UniquePtr<T>&& other) noexcept;
+
+	T& operator*() const;
+	T* operator->() const;
+private:
+	T* m_ptr;
+};
+
+template<typename T>
+UniquePtr<T>::UniquePtr(T* ptr) : m_ptr(ptr) {
+	cout << "UniquePtr ctor" << endl;
```

Comment:
The debug prints are nice for development stage but usually we don't commit them for CR